src/main/java/com/org/jenkins/custom/jenkins/distribution/service/GithubJwtHelper.java

```diff
@@ -0,0 +1,50 @@
+package com.org.jenkins.custom.jenkins.distribution.service;
+
+import io.jsonwebtoken.JwtBuilder;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.Jwts;
+import java.io.File;
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.PrivateKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Date;
+import com.google.common.io.Files;
+
+
+public class GithubJwtHelper {
+
+    static PrivateKey get(String filename) throws Exception {
+        byte[] keyBytes = Files.toByteArray(new File(filename));
+        PKCS8EncodedKeySpec spec = new PKCS8EncodedKeySpec(keyBytes);
+        KeyFactory kf = KeyFactory.getInstance("RSA");
+        return kf.generatePrivate(spec);
+    }
+
+    static String createJWT(String githubAppId, long ttlMillis) throws Exception {
+        //The JWT signature algorithm we will be using to sign the token
+        SignatureAlgorithm signatureAlgorithm = SignatureAlgorithm.RS256;
+
+        long nowMillis = System.currentTimeMillis();
+        Date now = new Date(nowMillis);
+
+        //We will sign our JWT with our private key
+        Key signingKey = get("custom_distribution_private_key.der");
```

Comment:
Should I push this too ?

Reply:
You shouldn't include secret keys unencrypted, so you'll want to encrypt the key in some way

Reply:
I am not too sure how to do that ? because this would be signed using rsa and the APP_ID so I am assuming if we keep the appID safe we can just push  this  ?